src/components/Nav/Nav.jsx

```diff
@@ -0,0 +1,63 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { Link, useLocation } from 'react-router-dom';
+import * as S from './Nav.styled';
+import UserLoggedIn from './UserLoggedIn';
+import logo from '../../assets/icons/logo.svg';
+
+const Nav = () => {
+  const [navFixedStatus, setNavFixedStatus] = useState(true);
+  const navRef = useRef();
+  const { pathname } = useLocation();
+
+  const currentPageIsFolderPage = useCallback(() => {
+    const isFolderPage = pathname === '/folder';
+    setNavFixedStatus(isFolderPage);
+    window.scrollTo(0, 0);
+  }, [pathname]);
+
+  const handleNavigation = () => {
+    if (!navRef.current) return;
+    const navNode = navRef.current;
+
+    if (window.scrollY > 30) {
+      navNode.classList.add('shadow');
+      return;
+    }
+
+    navNode.classList.remove('shadow');
+  };
+
+  useEffect(() => {
+    window.addEventListener('scroll', handleNavigation);
+    currentPageIsFolderPage();
+    return () => {
+      window.removeEventListener('scroll', handleNavigation);
+    };
+  }, [pathname, currentPageIsFolderPage]);
+
+  return (
+    <>
+      {navFixedStatus ? (
+        <S.HeaderLayout>
+          <S.LogoBox>
+            <Link to="/">
+              <img src={logo} alt="Linkbrary" />
+            </Link>
+          </S.LogoBox>
+          <UserLoggedIn />
+        </S.HeaderLayout>
+      ) : (
+        <S.HeaderLayout ref={navRef}>
```

Comment:
분기 처리한 이유가 ref 전달 차이만 있다면 아래처럼 작성도 가능한데요,
```suggestion
        <S.HeaderLayout ref={navFixedStatus ? navRef : undefined}>
```

ref 는 넣다뺐다 하지않고 prop 으로 전달한 상태로 쭉 둡니다 
```suggestion
        <S.HeaderLayout ref={navRef}>
```